### DISCUSSION TITLE
Can't figure out a GitHub Action

### DISCUSSION THREAD

**Discussion by osfanbuff63** [community/community#26669] [Actions]
<p>I can’t figure out why this GitHub Action will not work. My error says that every step needs to contain a <code>uses</code> or <code>run</code> key, but all of my steps have that. This is a rather long action, but here it is:</p>
<pre data-code-wrap="yml"><code class="lang-nohighlight">name: Publish release
on: 
  push:
    tags:
     - '**'
  workflow_dispatch:
jobs:
   create-release:
    name: Create release
    runs-on: ubuntu-latest
    steps:
     - name: Create release
       uses: ncipollo/release-action@v1.10.0
   upload-normal:
    name: Create non-optimized datapack
    runs-on: ubuntu-latest
    steps:
     - name: Clone repository
       uses: actions/checkout@v3
     - name: Zip files
       uses: vimtor/action-zip@v1
       with:
        files: './'
        dest: 'muffinhunt-datapack.zip'
     - name: Upload to release
       with:
        tag: ${{ github.ref }}
        generateReleaseNotes: true
   upload-optimized-datapack:
    name: Create and optimize datapack
    runs-on: ubuntu-latest
    steps:
     - name: Clone repository
       uses: actions/checkout@v3
       with:
          fetch-depth: 0
     - name: Run PackSquash
       uses: ComunidadAylas/PackSquash-action@master
       with:
          path: ./
          token: ${{ secrets.GITHUB_TOKEN }}
          minify_json_files: true
          validate_pack_metadata_file: true
          allow_optifine_mod: true
          delete_bloat_json_keys: true
          artifact_name: 'optimized-muffinhunt-datapack.zip'
     - name: Download optimized pack
       uses: actions/download-artifact@v3
       with:
         name: optimized-muffinhunt-datapack.zip
         path: ./
     - name: Publish optimized datapack
       uses: svenstaro/upload-release-action@v2
       with: 
         repo_token: ${{ secrets.GITHUB_TOKEN }}
         tag: ${{ github.ref }}
         file: optimized-muffinhunt-datapack.zip
   fetch-muffinhunt-resourcepack:
    name: Fetch latest MuffinHunt resource pack changes
    runs-on: ubuntu-latest
    steps:
     - name: Clone muffinhunt resource pack
       uses: actions/checkout@v3
       with: 
        repository: osfanmuffin/muffinhunt-resourcepack
        path: /resourcepack/
     - name: Zip files
       uses: vimtor/action-zip@v1
       with:
        files: /resourcepack/*
        dest: 'muffinhunt-resourcepack.zip'
     - name: Publish resource pack
       uses: svenstaro/upload-release-action@v2
       with: 
         repo_token: ${{ secrets.GITHUB_TOKEN }}
         tag: ${{ github.ref }}
         file: muffinhunt-resourcepack.zip
   optimize-muffinhunt-resourcepack:
    name: Fetch and optimize MuffinHunt resourcepack
    runs-on: ubuntu-latest
    steps:
     - name: Clone muffinhunt resource pack
       uses: actions/checkout@v3
       with: 
        repository: osfanmuffin/muffinhunt-resourcepack
        path: /resourcepack/
     - name: Run PackSquash
       uses: ComunidadAylas/PackSquash-action@master
       with:
          path: ./resourcepack/
          token: ${{ secrets.GITHUB_TOKEN }}
          minify_json_files: true
          validate_pack_metadata_file: true
          allow_optifine_mod: true
          delete_bloat_json_keys: true
          artifact_name: 'optimized-muffinhunt-resourcepack.zip'
     - name: Download optimized pack
       uses: actions/download-artifact@v3
       with:
         name: optimized-muffinhunt-resourcepack.zip
         path: ./
     - name: Publish resource pack
       uses: svenstaro/upload-release-action@v2
       with: 
         repo_token: ${{ secrets.GITHUB_TOKEN }}
         tag: ${{ github.ref }}
         file: optimized-muffinhunt-resourcepack.zip
</code></pre>
<p>The repo I am trying this on is <a href="https://github.com/osfanmuffin/muffinhunt-datapack" rel="noopener nofollow ugc">this one</a>.</p>
<p>Thanks!<br>
osfan</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="osfanbuff63" data-post="1" data-topic="242910">
<div class="title">
<div class="quote-controls"></div>
 osfanbuff63:</div>
<blockquote>
<p>My error says that every step needs to contain a <code>uses</code> or <code>run</code> key, but all of my steps have that.</p>
</blockquote>
</aside>
<p>The last one in the <code>upload-normal</code> job does not. I recommend you use a consistent indentation for YAML (I prefer 2 per level) and add empty lines between jobs, that makes it easier to see things.</p>

**Comment by osfanbuff63** [comment]
<p>I’m sorry, I forgot to mark an answer <a class="mention" href="/airtower-luna">@airtower-luna</a>! Sorry about that!</p>
